### PR TITLE
Automagically log out after 120 seconds of inactivity

### PR DESCRIPTION
**Pull request by reynders**
After logging in with PIN, 
- log out after 120 seconds of inactivity OR
- log out on blur (for example screen lock)
Tested to work on IOS 14 device

### CODE REVIEW

File: frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect } from 'react'
+
+export default function useAutoLogout(timeout: number, onTimeOut: () => void) {
+  const logout = useCallback(() => {
+    clearTimeout()
+    onTimeOut && onTimeOut()
+  }, [onTimeOut])
```
Comment (Joosakur) [inline frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx:7]: Abstraction level seems inconsistent. This takes onTimeOut callback which could be anything, but part of the naming implies that this hook is logout specific. I guess you can go either direction, but if the logout logic has to be passed in anyway and nothing inside this is really logout specific, I would probably remove mentions of logout and make this a generic useTimeOut or useInactivity hook.

File: frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    const logoutOnResume = () => {
+      clearTimeouts()
+      logout()
+    }
+
+    for (const event of events) {
+      window.addEventListener(event, resetTimeout, { passive: true })
+      window.addEventListener('blur', logoutOnResume, { passive: true })
```
Comment (Joosakur) [inline frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx:41]: should this be outside for loop? now blur handler gets added 7 times

File: frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx (hunk 1 above)
Comment (reynders) [reply frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx]: Renamed to useInactivityTimeout

File: frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx (hunk 2 above)
Comment (reynders) [reply frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx]: Good catch, fixed!

File: frontend/src/employee-mobile-frontend/components/attendances/child-info/InactivityTimeout.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  timeout: number,
+  onTimeOut: () => void
+) {
+  const logout = useCallback(() => {
+    clearTimeout()
+    onTimeOut && onTimeOut()
+  }, [onTimeOut])
+
+  useEffect(() => {
+    let logoutTimeout: number
```
Comment (Joosakur) [inline frontend/src/employee-mobile-frontend/components/attendances/child-info/InactivityTimeout.tsx:13]: vars on lines 7 and 13 could still be renamed

**Review by Joosakur — APPROVED** [review]

**Pull request merged**